odoo/tools/config.py

```diff
@@ -124,6 +124,9 @@ def __init__(self, fname=None):
         group.add_option("--upgrade-path", dest="upgrade_path",
                          help="specify an additional upgrade path.",
                          action="callback", callback=self._check_upgrade_path, nargs=1, type="string")
+        group.add_option("--pre-update-scripts", dest="pre_update_scripts", my_default="",
+                         help="Force specific upgrade scripts while loading base module (ignored unless -u)",
+                         action="callback", callback=self._check_scripts, nargs=1, type="string")
```

Comment:
Is it possible avoid adding an extra command?
Just an idea. use a dir for a dummy module `upgrade/migrations/__all__/saas~18.1/pre-migrate.py`
And we call 
1. `migrations.migrate_module(package, 'pre')` before upgradeing all modules
2. `migrations.migrate_module(package, 'post')` after loading all modules
3. `migrations.migrate_module(package, 'end')` after all `end-` of all modules (It could be used for some cleanup or check)
where `package` is a hack instance with `manifest`, `state` .... `name = '__all__'`

Reply:
I had something done in a different way see https://github.com/odoo/odoo/compare/16.0...odoo-dev:odoo:16.0-imp_force_scripts-afu?expand=1
* Use a new kind of scripts that could be distributed _directly_ by custom module devs
* They are loaded in the standard migration engine

After discussion the option in this PR was considered simpler. It is more direct and clear. With a CLI option devs only need to pass all custom scripts that may be needed from different sources. Despite "being useful", there is no _real_ need we could think of other than running in pre- in base. For running in post- a script in pre- in the custom module should be enough. Similarly for end-.

Reply:
Note that some of the downsides of having such extra scripts is that it becomes harder to decide _when_ they should be run. With an explicit flag this becomes clearer and less error prone.